aws/rust-runtime/aws-sigv4/src/http_request/sign.rs

```diff
@@ -93,6 +95,39 @@ pub enum SignableBody<'a> {
     StreamingUnsignedPayloadTrailer,
 }
 
+/// Formats the value using the given formatter. To print the body data, set `LOG_SIGNABLE_BODY=true`.
+impl<'a> Debug for SignableBody<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let should_log_signable_body = std::env::var(LOG_SIGNABLE_BODY)
+            .map(|v| v.eq_ignore_ascii_case("true"))
+            .unwrap_or_default();
+        match self {
+            Self::Bytes(arg0) => {
+                if should_log_signable_body {
+                    f.debug_tuple("Bytes").field(arg0).finish()
+                } else {
+                    f.debug_tuple("Bytes")
+                        .field(&"** REDACTED **. To print, set LOG_SIGNABLE_BODY=true")
+                        .finish()
+                }
+            }
+            Self::UnsignedPayload => write!(f, "UnsignedPayload"),
+            Self::Precomputed(arg0) => {
```

Comment:
I think you can log precomputed as the value should just be a checksum.

Reply:
Addressed in bba1a52